director/origin_api.go

```diff
@@ -260,18 +263,49 @@ func VerifyDirectorTestReportToken(strToken string) (bool, error) {
 	return false, nil
 }
 
-func GetRegistryIssuerURL(prefix string) (string, error) {
-	namespace_url_string := param.Federation_RegistryUrl.GetString()
-	if namespace_url_string == "" {
-		return "", errors.New("Namespace URL is not set")
+// For a given prefix, get the url of the issuer/public key from the registry
+// This works by looking up the namespace-configuration json for the namespace
+// and grabbing the value corresponding to the "jwks_uri" key.
+func GetNSIssuerURL(prefix string) (string, error) {
+	if prefix == "" || !strings.HasPrefix(prefix, "/") {
+		return "", errors.New(fmt.Sprintf("the prefix \"%s\" is invalid", prefix))
 	}
-	namespace_url, err := url.Parse(namespace_url_string)
+	registryUrlStr := param.Federation_RegistryUrl.GetString()
+	if registryUrlStr == "" {
+		return "", errors.New("federation registry URL is not set and was not discovered")
+	}
+	registryUrl, err := url.Parse(registryUrlStr)
 	if err != nil {
 		return "", err
 	}
-	namespace_url.Path, err = url.JoinPath(namespace_url.Path, "api", "v2.0", "registry", "metadata", prefix, ".well-known", "issuer.jwks")
+
+	registryUrl.Path, err = url.JoinPath(registryUrl.Path, "api", "v2.0", "registry", "metadata", prefix, ".well-known", "namespace-configuration")
```

Comment:
I think this should be changed to `url.JoinPath(registryUrl.Path, prefix, ".well-known", "openid-configuration")`

First because this is what's defined in [OIDC RFC](https://openid.net/specs/openid-connect-discovery-1_0.html#rfc.section.4) that the `./well-known/openid-configuration` MUST be concatenated directly to the hostname, and the name must be `/openid-configuration`

Fortunately, per OIDC spec, we _can_ append a path before the `./well-known` for multi-tenant use case where a single issuer can serve multiple tenants based on their requested path prefix. It fits perfect into our scenario that we can append the namespace prefix before `./well-known`. It should look like `https://registry.com:9999/test/prefix/.well-known/openid-configuration`.

To do the route matching, we can first check is the path `HasSuffix` with `/.well-known/openid-configuration`, then check if it doesn't have a prefix `/api` or `/view` or any other static resources. If it does, then it's not meant for this purpose. 

The caveat for this is that we will unfortunately hit the `wildcard hell` at the top level where we need to handle the wildcard matching plus all additional routes under root path in the wildcard handler like we did [here](https://github.com/PelicanPlatform/pelican/blob/3caae35f06cd96ae97c2afca51a470b98c060010/registry/registry.go#L646)